### PR TITLE
Popup login screen the first time only

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

 A little trickier than first glance.

Now we have a function called `isFirstTime` that returns a `Promise<boolean>`.

Login screen (and kbfs splash page, like before) only show up the first time.

### CODE REVIEW

File: desktop/app/index.js [hunk 1]
```diff
@@ -55,9 +53,13 @@ installer(err => {
   splash()
 })
 
-if (app.dock && !mw.initiallyVisible) {
-  app.dock.hide()
-}
+app.once('ready', () => {
+  const mw = mainWindow()
```
Comment (MarcoPolo) [inline desktop/app/index.js]: There's a potential race condition where we try to show the main window before the app is ready.

File: desktop/app/main-window.js [hunk 2]
```diff
@@ -14,6 +15,17 @@ export default function () {
     }
   )
 
+  isFirstTime().then(firstTime => {
+    if (firstTime) {
+      mainWindow.show(true)
+      mainWindow.window.webContents.once('did-finish-load', () => {
```
Comment (MarcoPolo) [inline desktop/app/main-window.js]: in case the window isn't ready to be shown (like the bug on windows)

File: desktop/app/first-time.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import {splashRoot} from '../shared/constants/platform.native.desktop'
+import path from 'path'
+import fs from 'fs'
+
+const filePath = path.join(splashRoot, 'started.txt')
+
+let _firstTimeCachedValue = null
+
+let isFirstTime: () => Promise<boolean> = () => new Promise((resolve, reject) => {
+  resolve(true)
```
Comment (chrisnojima) [inline desktop/app/first-time.js]: ???????

Comment (MarcoPolo) [inline desktop/app/first-time.js]: ahh testing!

File: desktop/app/first-time.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// Checks to see if we've launched before (we touch started.txt). If not, write the file and open a browser
+import {splashRoot} from '../shared/constants/platform.native.desktop'
+import path from 'path'
+import fs from 'fs'
+
+const filePath = path.join(splashRoot, 'started.txt')
+
+let _firstTimeCachedValue = null
+
+let isFirstTime: () => Promise<boolean> = () => new Promise((resolve, reject) => {
```
Comment (chrisnojima) [inline desktop/app/first-time.js]: why is this even a function? can't we just have the promise and return it and get rid of the cachedValue (aka cached value is the promise)

Comment (chrisnojima) [inline desktop/app/first-time.js]: other than that 👍

Comment (MarcoPolo) [inline desktop/app/first-time.js]: Didn't realize promises worked that way!!

Comment (chrisnojima) [inline desktop/app/first-time.js]: yeah so its worth reading stuff like http://www.mattgreer.org/articles/promises-in-wicked-detail/

**Pull request merged**